### DISCUSSION TITLE
Setting the Kvantum configuration flexibly (maybe FR)

### DISCUSSION THREAD

**Discussion by florhizome** [tsujan/Kvantum#688] [General]
Hello, 
I use Kvantum across multiple DEs I have installed - most prominently LXQT and a custom desktop around Wayfire.
I would like to have distinct configurations/themes between those, but I don't really see a documented way, of how that could be done. Basically I don't want to mess up my LXQT configuration, when I'm changing stuff in the more experimental Wayfire environment.  Am I overlooking something there?
A way I would imagine this to work could be via environmental variable that can be set, for my usecase via wrapper script. something like `KVANTUM_THEME` or `KVANTUM_CONFIG`. But that's just an idea - anyway that makes the afroementioned functionality available would be great!.
I'm not sure how that could best be integrated with the way kvantum manager works, etc.

**Comment by tsujan** [comment]
Kvantum checks several places to find its config files by giving each a priority — no other style engine is more flexible when it comes to config files. It can also give different themes to different apps and has a "dark" variant switch (which seems redundant to me now — it was added for Gnome and I might remove it at some point). All this has had a cost, regarding code simplicity and readability.

Adding more config places/files or defining EVs would cost even more. I'm afraid it can't be justified by the special use case you mentioned.

**Reply by florhizome** [reply]
While I describe my usecase kinda specific, i don't think the change, if achieved, is that specific.
Kvantum (+Qt5ct) is already a handy tool for users to theme their Qt applications in a Desktop-agnostic way. Not being able to automate the selection of themeing myself is a pretty big boon to that, though. Why would anyone want to be bound to use the same Qt theme on every of their desktops? This severly narrows the potential usefulness of kvantum IMO. it could be the defacto generic tool to theme Qt on linux - Or you are limited to one theme across all desktops, which means you basically can only have one usecase/approach to use Kvantum for. 
I sincerely wouldn't enjoy LXQT as much, if I couldn't use many Kvantum themes that are actually designed with KDE in mind. I don't think LXQT's or any lesser Qt desktop's native themeing engine will be able to compete with that anytime soon, so a cross-desktop themeing tool is important for those to keep up with the possibilities of the bigger DEs - given it is flexible enough, to be used in such a way.
There probably can be various way of implementing this. I don't really want to go to deep into prescribing ideas, that i could not implement, though. So I would be ok with just leaving this open, so other people can have a though about it, if you will. I understand if you don't want to write it yourself.

**Comment by tsujan** [comment]
Kvantum doesn't have an attachment to LXQt, KDE or any other DE. Only by chance, I'm also an LXQt developer.

Users usually have one main DE and apps usually have the same settings under all DEs.

**Reply by florhizome** [reply]
That's OK. 
If you say so. 

Good Luck with the App i guess, i still like it. Not sure what to contribute to this discussion any further.
